addons/html_editor/static/src/main/list/list_plugin.js

```diff
@@ -127,6 +134,44 @@ export class ListPlugin extends Plugin {
         }
     }
 
+    onInput() {
+        const doc = this.editable.ownerDocument;
+        const selection = doc.getSelection();
+        const blockEl = closestBlock(selection.anchorNode);
+        const stringToConvert = blockEl.textContent.substring(0, selection.anchorOffset);
+        const shouldCreateNumberList = /^(?:[1aA])[.)]\s$/.test(stringToConvert);
+        const shouldCreateBulletList = /^[-*]\s$/.test(stringToConvert);
+        if (
+            (shouldCreateNumberList || shouldCreateBulletList) &&
+            !closestElement(selection.anchorNode, "li")
+        ) {
+            const range = selection.getRangeAt(0);
+            range.setStartBefore(blockEl.firstChild);
+            range.extractContents();
+            fillEmpty(blockEl);
+            if (shouldCreateNumberList) {
+                this.dispatch("TOGGLE_LIST", { mode: "OL" });
+                // When the anchorNode is a context block and a list is
+                // being created inside it, ensure to navigate to the
+                // deepest node.
+                const [deepsetNode] = getDeepestPosition(
+                    selection.anchorNode,
+                    selection.anchorOffset
+                );
+                this.shared.disableObserver();
```

Comment:
Why do we want to disable the observer ? I think we want to propagate this to the history for collaborative purpose. Test this use case:

- Client A: type `A. Hello world`
- Client B: see `1. Hello world`

Reply:
@fdardenne When creating a list using `A. `, it will initially create a simple numbered list. Afterwards, applying the `upper-alpha` style will convert it to an `A.` list. 
If I don't disable the observer, performing `ctrl + z` will undo the `A.` list back to a `1.` list. This happens because the observer records a step: transforming the `1.` list into an `A.` list. To prevent this step from being recorded, I have disable the observer. This ensures that when `ctrl + z` is performed from the `A.` list, it will convert it back to plain `A. ` text.

Reply:
Thanks for the explanation. Indeed it's not easy to manage it since `this.dispatch("TOGGLE_LIST");` does an `ADD_STEP`.

What you can do is this:
```diff
@@ -119,6 +119,7 @@ export class ListPlugin extends Plugin {
         switch (command) {
             case "TOGGLE_LIST":
                 this.toggleList(payload.mode);
+                this.dispatch("ADD_STEP");
                 break;
             case "NORMALIZE": {
                 this.normalize(payload.node);
@@ -195,8 +196,6 @@ export class ListPlugin extends Plugin {
                 this.liToBlocks(li);
             }
         }
-
-        this.dispatch("ADD_STEP");
     }

     normalize(root = this.editable) {
```

And in `onInput` replace `this.dispatch("TOGGLE_LIST")` by directly calling the `this.toggleList` method. It's not an ideal solution but it will fix the problems.

After your PR, we will try to find a way to improve the add_step to avoid this kind of problem in the future.

Reply:
@fdardenne Applied the requested changes.

Also, found sone other bugs in collaboration when using list in master.

**First:**
-   Type a some word, `Hello World`.
-   Create a list.
-   Client A: see `1. Hello World`
-   Client B: see `1.` (the text disappears, leaving only empty list).

**Second:**

- Create an `H1` tag and type "Hello World".
- Create a list.
-  For client A The HTML structure is `<ol><li><h1>Hello World</h1></li></ol>`.
- For client B The HTML structure is `<h1>Hello World</h1><ol><li></li></ol>` (the list is empty and the H1 tag is not removed).